WordPress/Classes/Networking/ServiceRemoteREST.h

```diff
@@ -28,6 +34,15 @@ typedef enum
  */
 - (id)initWithApi:(WordPressComApi *)api;
 
+/**
+ *  @brief      Designated initializer.
+ *
+ *  @param      api     The API to use for communication.  Cannot be nil.
+ *
+ *  @returns    The initialized object.
+ */
+- (id)initWithComRestApi:(WordPressComRestApi *)api;
```

Comment:
I don't think we need a new base class because when me migrate all the services we will remove the api property and just keep the new property, so this will be a transition phase until we move all the services.